pkg/tests/tasks/traffic/ingress/gatewayapi_test.go

```diff
@@ -0,0 +1,205 @@
+        spec:
+          runtime:
+            components:
+              pilot:
+                container:
+                  env:
+                    PILOT_ENABLE_GATEWAY_API: “true”
+                    PILOT_ENABLE_GATEWAY_API_STATUS: “true”
+                    PILOT_ENABLE_GATEWAY_API_DEPLOYMENT_CONTROLLER: “true”`)
+
+				t.Cleanup(func() {
+					oc.Patch(t, meshNamespace, "smcp", SMCP_NAME, "json",
+						`[{"op": "remove", "path": "/spec/runtime"}]`)
+				})
+
+			} else {
+				oc.Patch(t, meshNamespace, "smcp", SMCP_NAME, "merge", `
+        spec:
+          techPreview:
+            gatewayAPI:
+              enabled: true`)
+
+				t.Cleanup(func() {
+					oc.Patch(t, meshNamespace, "smcp", SMCP_NAME, "json",
+						`[{"op": "remove", "path": "/spec/techPreview"}]`)
+				})
+
+			}
+
+			t.LogStep("Deploy the Gateway API configuration including a single exposed route (i.e., /get)")
+			oc.ApplyTemplate(t, ns, gatewayAndRouteYAML, map[string]string{"GatewayClassName": "istio"})
+			t.Cleanup(func() {
+				oc.DeleteFromTemplate(t, ns, gatewayAndRouteYAML, map[string]string{"GatewayClassName": "istio"})
+			})
+
+			t.LogStep("Wait for Gateway to be ready")
+			oc.WaitCondition(t, ns, "Gateway", "gateway", "Ready")
+
+			t.LogStep("Verfiy the GatewayApi access the httpbin service using curl")
+			retry.UntilSuccess(t, func(t TestHelper) {
+				oc.Exec(t,
+					pod.MatchingSelector("app=istio-ingressgateway", meshNamespace),
+					"istio-proxy",
+					"curl http://gateway.foo.svc.cluster.local:8080/get -H Host:httpbin.example.com -s -o /dev/null -w %{http_code}",
+					assert.OutputContains("200",
+						"Access the httpbin service with GatewayApi",
+						"Unable to access the httpbin service with GatewayApi"))
+			})
+		})
+
+		t.NewSubTest("Deploy the Gateway-Controller Profile").Run(func(t test.TestHelper) {
+
+			t.LogStep("Install httpbin")
+			app.InstallAndWaitReady(t, app.Httpbin(ns))
+
+			t.LogStep("Deploy SMCP with the profile")
+			oc.ApplyTemplate(t,
+				meshNamespace,
+				gatewayControllerProfile,
+				map[string]string{"Name": "basic", "Version": env.GetSMCPVersion().String()})
+			oc.WaitSMCPReady(t, meshNamespace, "basic")
+
+			t.LogStep("delete default SMMR and create custom SMMR")
+			oc.DeleteFromString(t, meshNamespace, defaultSMMR)
+			oc.ApplyTemplate(t, meshNamespace, createSMMR, map[string]string{"Member": ns})
+			oc.WaitSMMRReady(t, meshNamespace)
+
+			t.LogStep("Deploy the Gateway API configuration including a single exposed route (i.e., /get)")
+			oc.ApplyTemplate(t, ns, gatewayAndRouteYAML, map[string]string{"GatewayClassName": "ocp"})
+			t.Cleanup(func() {
+				oc.DeleteFromTemplate(t, ns, gatewayAndRouteYAML, map[string]string{"GatewayClassName": "ocp"})
+			})
+
+			t.LogStep("Wait for Gateway to be ready")
+			oc.WaitCondition(t, ns, "Gateway", "gateway", "Ready")
+
+			t.LogStep("Verfiy the GatewayApi access the httpbin service using curl")
```

Comment:
Misspell here, should be Verify and we need to add the expected result here at the end. `Verify the GatewayApi access to the httpbin service using curl. Expected to succeed`